File: svnrepo/RIT_JBF_source/src/PlayE2AnimationButton.java
/*
 * Copyright (C) 2013 Rochester Institute of Technology
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 2 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 *
 * See the GNU General Public License for more details.
 */

/**
 *
 * @author Amanda Fisher
 */

import javax.swing.JButton;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class PlayE2AnimationButton extends JButton implements ActionListener {

    Electro2D electro2D;
    boolean playing;
    private boolean iefDrawn;
    private boolean sdsDrawn;
    private static boolean compareFiles;
    String choice;
    boolean isPlaying;

    public PlayE2AnimationButton(Electro2D e) {

        super("Run");

        String hoverText;
        if(iefDrawn == false){
            hoverText = "Run IEF animation";
            }else{
                hoverText = "Run SDS Page animation";
            }
        super.setToolTipText(hoverText);
        addActionListener(this);
        electro2D = e;
        playing = false;
        iefDrawn = false;
        sdsDrawn = false;
        compareFiles = false;
    }

    public boolean getSdsStatus(){
	return sdsDrawn;
    }

    public void resetPlay(){
	playing = false;
        this.setText("Run");
    }

    public void resetIEF(){
	iefDrawn = false;
    }

    public void resetSdsStatus(){
	sdsDrawn = false;
    }

    public static void setCompare(boolean bool){
	compareFiles = bool;
    }

    public void actionPerformed(ActionEvent e) {
        
        if(playing == true) {
	    playing = false;
	    electro2D.stopThread();
	}
	// otherwise determine which parts of the animation need to be drawn
	// and start the appropriate thread.
	else {
	    //get the animation the user wishes to see
            electro2D.clearpH();
	    choice = electro2D.getAnimationChoice();
	    if( electro2D.getSequencesReady()){

		//if the data for the animation needs to be processed,
		//do so
		    if( choice.equals( "IEF" )){
			electro2D.getGel().prepare();
			electro2D.resetBool();
                        if( compareFiles ){
			    electro2D.getGel().prepare2();
			}
		    }

		//if the user selected IEF animation and the image is not
		// already displayed on the screen, perform the IEF
		// animation
		if( choice.equals( "IEF") ){
		    if( iefDrawn == false ){
			electro2D.restartIEF();
			iefDrawn = true;
			electro2D.getGel().resetReLine();
		    }
		}
		//if the user selected SDS-PAGE animation, and the IEF is
		// already drawn, perform the SDS-PAGE animation
		else if( choice.equals( "SDS-PAGE" )){
                    if( iefDrawn == true ){
			electro2D.getGel().clearCanvas();
                        electro2D.getGel().clearIEF();
			electro2D.getGel().resetLocation();
                        electro2D.clearpH();
			playing = true;
			electro2D.restartThread();
			sdsDrawn = true;
		    }
		}
	    }
	}

        // let the user know whether they can pause or play the animation
        if(playing == false) {
	    this.setText("Run");
	} else {
	    this.setText("Pause");
	}

    }
}

File: site/swingjs/j2s/main/java/Electro2D/PlayE2AnimationButton.js
(function(){var P$=Clazz.newPackage("main.java.Electro2D"),I$=[[0,'main.java.Electro2D.ColorFrame']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "PlayE2AnimationButton", null, 'javax.swing.JButton', 'java.awt.event.ActionListener');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);

C$.$fields$=[['Z',['sdsPlaying','iefDrawn','sdsDrawn'],'S',['choice'],'O',['electro2D','main.java.Electro2D.Electro2D']]
,['Z',['compareFiles']]]

Clazz.newMeth(C$, 'c$$main_java_Electro2D_Electro2D',  function (e) {
;C$.superclazz.c$$S.apply(this,["Start"]);C$.$init$.apply(this);
this.addActionListener$java_awt_event_ActionListener(this);
this.electro2D=e;
this.sdsPlaying=false;
this.iefDrawn=false;
this.sdsDrawn=false;
C$.compareFiles=false;
this.setHoverText$();
}, 1);

Clazz.newMeth(C$, 'getSdsStatus$',  function () {
return this.sdsDrawn;
});

Clazz.newMeth(C$, 'resetPlay$',  function () {
this.sdsPlaying=false;
this.setText$S("Start");
});

Clazz.newMeth(C$, 'resetIEF$',  function () {
this.iefDrawn=false;
});

Clazz.newMeth(C$, 'resetSdsStatus$',  function () {
this.sdsDrawn=false;
});

Clazz.newMeth(C$, 'setCompare$Z',  function (bool) {
C$.compareFiles=bool;
}, 1);

Clazz.newMeth(C$, 'setHoverText$',  function () {
if (this.sdsPlaying) {
C$.superclazz.prototype.setText$S.apply(this, ["Stop"]);
this.setToolTipText$S("Stop separation");
} else if (!this.iefDrawn && !this.sdsDrawn ) {
C$.superclazz.prototype.setToolTipText$S.apply(this, ["Start IEF pH separation"]);
} else if (this.iefDrawn && !this.sdsDrawn ) {
C$.superclazz.prototype.setToolTipText$S.apply(this, ["Start SDS-Page separation"]);
} else if (this.iefDrawn && this.sdsDrawn && !this.sdsPlaying  ) {
C$.superclazz.prototype.setToolTipText$S.apply(this, ["Press \'Restart\' to start over."]);
} else {
C$.superclazz.prototype.setToolTipText$S.apply(this, ["Press \'Add Proteins\' to begin"]);
}});

Clazz.newMeth(C$, 'actionPerformed$java_awt_event_ActionEvent',  function (e) {
if (this.sdsPlaying) {
C$.superclazz.prototype.setText$S.apply(this, ["Stop"]);
C$.superclazz.prototype.setToolTipText$S.apply(this, ["Stop Separation"]);
this.electro2D.stopThread$();
this.sdsPlaying=false;
} else {
this.electro2D.clearpH$();
this.choice=this.electro2D.getAnimationChoice$();
if (this.electro2D.getSequencesReady$()) {
if (this.choice.equals$O("IEF")) {
this.electro2D.getGel$().prepare$();
this.electro2D.resetBool$();
if (C$.compareFiles) {
this.electro2D.getGel$().prepare2$();
}}if (this.choice.equals$O("IEF")) {
if (!this.iefDrawn) {
this.electro2D.restartIEF$();
this.iefDrawn=true;
this.electro2D.getGel$().setRedrawPHLines$Z(false);
}} else if (this.choice.equals$O("SDS-PAGE")) {
if (this.iefDrawn) {
this.electro2D.getGel$().resetLocation$();
this.electro2D.clearpH$();
this.sdsPlaying=true;
this.electro2D.restartThread$();
this.sdsDrawn=true;
Clazz.new_($I$(1,1)).showKey$();
}}this.setHoverText$();
}}});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('3.3.1-v4');//Created 2022-03-22 08:41:08 Java2ScriptVisitor version 3.3.1-v4 net.sf.j2s.core.jar version 3.3.1-v4
